tests/test_output.py

```diff
@@ -894,6 +895,57 @@ def test_learning_curve_ylimits():
     assert_almost_equal(ylimits_dict['r2'][1], 0.67, decimal=2)
 
 
+@raises(ValueError)
+def test_learning_curve_min_examples_check():
+    """
+    Test to check learning curve raises error with less than 500 examples
+    """
+    # generates a training split with less than 500 examples
+    train_fs_less_than_500, _ = make_classification_data(num_examples=499,
+                                                         train_test_ratio=1.0,
+                                                         num_labels=3)
+
+    # creating an example learner
+    learner = Learner('LogisticRegression')
+
+    # this must throw an error because `examples` has less than 500 items
+    _ = learner.learning_curve(examples=train_fs_less_than_500, metric='accuracy')
+
+
+def test_learning_curve_min_examples_check_override():
+    """
+    Test to check learning curve displays warning with less than 500 examples
+    """
+
+    # creates a logger which writes to a temporary log file
+    log_file_path = join(_my_dir, "output",
+                         "test_learning_curve_min_examples_check_override.log")
+    logger = get_skll_logger("test_learning_curve_min_examples_check_override",
```

Comment:
Once this logger is no longer need, it needs to be closed (and the file handle needs to be closed). Use this function to do that:

```python
from skll.utils.logging import close_and_remove_logger_handlers

close_and_remove_logger_handlers(logger)
```

This would be done at the end of the test (after `learner.learning_curve` is called at least).